internal/controllers/machineset/machineset_controller.go

```diff
@@ -362,8 +362,21 @@ func (r *Reconciler) syncMachines(ctx context.Context, machineSet *clusterv1.Mac
 	log := ctrl.LoggerFrom(ctx)
 	for i := range machines {
 		m := machines[i]
-		// If the machine is already being deleted, we don't need to update it.
+		// If the machine is already being deleted, we only need to sync
+		// the subset of fields that impact tearing down a machine
 		if !m.DeletionTimestamp.IsZero() {
+			patch := client.MergeFrom(m.DeepCopy())
```

Comment:
Let's please use the patchHelper here (see e.g. l.1010), it's doing basically the same here, but I would prefer using it so we can profit from error handling & potential improvements here as well (e.g. logging etc.)

Reply:
fixed